streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -244,47 +247,41 @@ public void shouldCreateProducer() {
     @Test
     public void shouldForwardCallToPartitionsFor() {
         final List<PartitionInfo> expectedPartitionInfo = Collections.emptyList();
-        expect(mockedProducer.partitionsFor("topic")).andReturn(expectedPartitionInfo);
-        replay(mockedProducer);
+        when(mockedProducer.partitionsFor(topic)).thenReturn(expectedPartitionInfo);
 
         final List<PartitionInfo> partitionInfo = streamsProducerWithMock.partitionsFor(topic);
 
         assertThat(partitionInfo, sameInstance(expectedPartitionInfo));
-        verify(mockedProducer);
+        verify(mockedProducer).partitionsFor(topic);
```

Comment:
This is not necessary, the StrictStubs make certain to check all `when`s you have used have indeed been invoked.